src/container.cpp

```diff
@@ -682,13 +682,22 @@ void Container::postRemoveNotification(Thing* thing, const Cylinder* newParent,
 
 void Container::internalAddThing(Thing* thing) { internalAddThing(0, thing); }
 
-void Container::internalAddThing(uint32_t, Thing* thing)
+void Container::internalAddThing(uint32_t index, Thing* thing)
 {
 	Item* item = thing->getItem();
 	if (!item) {
 		return;
 	}
 
+	if (index == 0 || itemlist.empty()) {
+		itemlist.push_front(item);
+	} else {
+		size_t pos = std::min<size_t>(std::max<size_t>(0, itemlist.size() - 1), index);
+		auto it = itemlist.begin();
+		std::advance(it, pos);
```

Comment:
You don't need to check the bounds when advancing from `begin()`, so

```suggestion
		auto it = itemlist.begin() + index;
```

Also works. You can also drop the `if` here, since `push_front(item)` and `insert(begin(), item)` are equivalent.

Reply:
Fixed!